src/Lecture1.hs

```diff
@@ -108,7 +135,12 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str = numsum
+    where
+        numstrings = words str
+        numints = map (\x -> read x::Int) numstrings 
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
        numints = map read numstrings
```

You can even implement `strSum` as a simple one-liner:

```haskell
strSum str = sum (map read (words str))
```